src/main/java/net/snowflake/ingest/SimpleIngestManager.java

```diff
@@ -322,7 +411,7 @@ public SimpleIngestManager(
    * @throws NoSuchAlgorithmException if can't create key factory by using RSA algorithm
    * @throws InvalidKeySpecException if private key or public key is invalid
    */
-  private KeyPair createKeyPairFromPrivateKey(PrivateKey privateKey)
+  public static KeyPair createKeyPairFromPrivateKey(PrivateKey privateKey)
```

Comment:
Was this made public for testing? Can we make it @ VisibleForTesting instead?

Reply:
Technically yes, but I thought it could be used as well hence made it public static.

Reply:
Also, I dont know why all keypair APIs were deprecated but it seems like they can be used. Builder has setKeyPair which would mean they can use this public API.

Reply:
Here is what I propose:
1. I will keep the code as is. (I know it is all over the place)
2. I will change keypair extraction logic and generting it inside SecurityManager.. 
3. We can provide two methods: generateP8Key from privateKey bytes and also generateKPFromP8Key - Both of them can be useful for clients. (Helper methods)